include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -192,7 +192,7 @@ class __kernel_name_base
     __compile_kernel(_Exec&& __exec)
     {
 #if _ONEDPL_KERNEL_BUNDLE_PRESENT
-        auto __kernel_bundle = sycl::get_kernel_bundle(__exec.queue().get_context());
+        auto __kernel_bundle = sycl::get_kernel_bundle<sycl::bundle_state::object>(__exec.queue().get_context());
```

Comment:
Could you please clarify why did you choose `object`, not `executable`?
SYCL 2020:
>All kernels defined in the SYCL application, however, must be exposed in bundle_state::executable because this is the only state that allows a kernel to be invoked on a device. Device built-in kernels are also exposed in 
bundle_state::executable.

>/* Available only when: (State == bundle_state::executable) */
kernel get_kernel(const kernel_id &kernelId) const;

+ @romanovvlad , any objections for usage?

Reply:
actually, I'm sure about 'object'... I saw the test https://github.com/intel/llvm/blob/54ede407edeb93b7e9334d1e725f48bf981b2965/sycl/test/on-device/basic_tests/kernel_bundle/kernel_bundle_api.cpp#L338

Reply:
.. Although I read SYCL spec, I have not catch the difference between `input`, `object `and  `executable` states...
Probably, Vlad will clarify that...

Reply:
<!--StartFragment-->
bundle_state::input | The device images in the kernel bundle have a format       that must be compiled and linked before their kernels can be invoked.       For example, an implementation could use this state for device images       that are stored in an intermediate language format or for device images       that are stored as source code strings.
-- | --
bundle_state::object | The device images in the kernel bundle have a format       that must be linked before their kernels can be invoked.
bundle_state::executable | The device images in the kernel bundle are in a format       that allows them to be invoked on a device.  For example, an       implementation could use this state for device images that have been       compiled into the device’s native code.

<!--EndFragment-->

Reply:
this is from the spec, we can think of them as `input` == ``.cpp`, `object` == `.o`, `executable` == final executable which can be run.

Reply:
@romanovvlad,
Not sure I understand analogy. What is your opinion what is better for our case? Or maybe it depends on how it's compiled by users?

Reply:
I'm afraid, additionally we need to use `use_kernel_bundle` method inside the `submit` and remove the kernel argument from `parallel_for`...

Reply:
>> What is your opinion what is better for our case? 

`get_kernel_bundle<executable>`
should be equivalent to 
```
sycl::program __program(__exec.queue().get_context());
__program.build_with_kernel_type<_DerivedKernelName>();
```

Reply:
However, Lueck, Gregory wrote: 
"...  In order to launch a kernel that you get from a kernel bundle, you should call “handler::use_kernel_bundle()” instead."

Reply:
While the proper way of usage of  “handler::use_kernel_bundle()” is under discussion, I'm going to revert the code which uses one.